File: E-Tour/src/main/java/com/example/demo/pricing/Occupancy.java
package com.example.demo.pricing;

public final class Occupancy {

    public static final int MAX_PER_ROOM = 2;
    public static final int MAX_EXTRA_BEDS_PER_ROOM = 1;
    public static final int MAX_OCCUPANTS_PER_ROOM = MAX_PER_ROOM + MAX_EXTRA_BEDS_PER_ROOM;

    public static final String TWIN_SHARING = "TWIN_SHARING";
    public static final String SINGLE = "SINGLE";
    public static final String EXTRA_PERSON = "EXTRA_PERSON";

    private Occupancy() {
    }

    public static String defaultOccupancy() {
        return TWIN_SHARING;
    }

    public static int roomsRequired(int twinSharingCount, int singleCount, int extraBedCount) {
        int twinRooms = (int) Math.ceil(twinSharingCount / (double) MAX_PER_ROOM);
        int extraBedCapacity = twinRooms * MAX_EXTRA_BEDS_PER_ROOM;
        int overflowBeds = Math.max(0, extraBedCount - extraBedCapacity);
        return twinRooms + singleCount + overflowBeds;
    }
}

File: ETour-DotNet/Pricing/Occupancy.cs
namespace ETour.Api.Pricing;

public static class Occupancy
{
    public const int MaxPerRoom = 2;
    public const int MaxExtraBedsPerRoom = 1;
    public const int MaxOccupantsPerRoom = MaxPerRoom + MaxExtraBedsPerRoom;

    public const string TwinSharing = "TWIN_SHARING";
    public const string Single = "SINGLE";
    public const string ExtraPerson = "EXTRA_PERSON";

    public static string DefaultOccupancy() => TwinSharing;

    public static int RoomsRequired(int twinSharingCount, int singleCount, int extraBedCount)
    {
        var twinRooms = (int)Math.Ceiling(twinSharingCount / (double)MaxPerRoom);
        var extraBedCapacity = twinRooms * MaxExtraBedsPerRoom;
        var overflowBeds = Math.Max(0, extraBedCount - extraBedCapacity);
        return twinRooms + singleCount + overflowBeds;
    }
}
